### PR TITLE
Replace test methods with lambdas in sparksql/tests/StringTest.cpp

### LINKED ISSUE #9065: Make test methods lambdas in sparksql/tests/StringTest.cpp
### Description

Make class methods using by tests in StringTest.cpp lambdas within the test.

### PR DESCRIPTION
**Pull request by acvictor**
Make class methods using by tests in StringTest.cpp lambdas within the test.

Fixes #9065

### CODE REVIEW

**Comment by acvictor** [comment]
@PHILO-HE @rui-mo I have raised this PR to address a comment from PR #8945. Can you please review?

File: velox/functions/sparksql/tests/StringTest.cpp [hunk 1]
```diff
@@ -21,213 +21,17 @@
...
-      std::optional<std::string> strArray) {
-    return evaluateOnce<int32_t>("find_in_set(c0, c1)", str, strArray);
-  }
+  // This is a five codepoint sequence that renders as a single emoji.
+  static constexpr char kWomanFacepalmingLightSkinTone[] =
+      "\xF0\x9F\xA4\xA6\xF0\x9F\x8F\xBB\xE2\x80\x8D\xE2\x99\x80\xEF\xB8\x8F";
 };
 
 TEST_F(StringTest, ascii) {
+  auto ascii = [&](std::optional<std::string> arg) {
```
[the remark this answers is not in the document]
Comment (acvictor) [reply velox/functions/sparksql/tests/StringTest.cpp]: Done, thanks!

File: velox/functions/sparksql/tests/StringTest.cpp [hunk 2]
```diff
@@ -290,7 +97,140 @@ TEST_F(StringTest, chr) {
...
+
+  // Null result.
+  EXPECT_EQ(conv("", 10, 16), std::nullopt);
+  EXPECT_EQ(conv(" ", 10, 16), std::nullopt);
+  EXPECT_EQ(conv("", std::nullopt, 16), std::nullopt);
+  EXPECT_EQ(conv("", 10, std::nullopt), std::nullopt);
+}
+
+TEST_F(StringTest, endsWith) {
+  auto endsWith = [&](const std::optional<std::string>& str,
```
[the remark this answers is not in the document]
Comment (acvictor) [reply velox/functions/sparksql/tests/StringTest.cpp]: Updated, thanks!

File: velox/functions/sparksql/tests/StringTest.cpp [hunk 3]
```diff
@@ -322,21 +281,299 @@ TEST_F(StringTest, lengthString) {
...
+  EXPECT_EQ(length(""), 0);
+  EXPECT_EQ(length(std::string("\0", 1)), 1);
+  EXPECT_EQ(length("1"), 1);
+  EXPECT_EQ(length("😋"), 4);
+  EXPECT_EQ(length(kWomanFacepalmingLightSkinTone), 17);
+  EXPECT_EQ(length("1234567890abdef"), 15);
+}
+
+TEST_F(StringTest, lpad) {
+  std::string invalidString = "Ψ\xFF\xFFΣΓΔA";
```
Comment (PHILO-HE) [inline velox/functions/sparksql/tests/StringTest.cpp:297]: Nit: const

File: velox/functions/sparksql/tests/StringTest.cpp [hunk 4]
```diff
@@ -322,21 +281,299 @@ TEST_F(StringTest, lengthString) {
...
+  std::string invalidPadString = "\xFFΨ\xFF";
```
Comment (PHILO-HE) [inline velox/functions/sparksql/tests/StringTest.cpp:298]: Ditto.

**Review by PHILO-HE — COMMENTED** [review]
@acvictor, thanks for your work! Just two small suggestion.

File: velox/functions/sparksql/tests/StringTest.cpp (hunk 3 above)
Comment (acvictor) [reply velox/functions/sparksql/tests/StringTest.cpp]: Updated, thank you!

File: velox/functions/sparksql/tests/StringTest.cpp (hunk 4 above)
Comment (acvictor) [reply velox/functions/sparksql/tests/StringTest.cpp]: Updated, thanks!

**Comment by acvictor** [comment]
@mbasmanova I have refactored this test suite as per your comments from this earlier PR - https://github.com/facebookincubator/velox/pull/8945. Thanks!

**Review by mbasmanova — APPROVED** [review]
@acvictor Nice refactoring. Thanks.

**Review by Yuhta — APPROVED** [review]

**Pull request closed without merging**